scripts/showcase/samplesheet-by-override-cycles.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/env python3
+
+"""
+Take in a samplesheet,
+Look through headers, rename as necessary
+Look through samples, update override cycles logic as necessary
+Split samplesheet out into separate override cycles files
+Write to separate files
+"""
+
+# Imports
+import re
+import os
+import pandas as pd
+import logging
+import argparse
+from pathlib import Path
+import sys
+
+# Set logging level
+logging.basicConfig(level=logging.DEBUG)
+
+"""
+Read:
+config = configparser.RawConfigParser()
+config.read('example.cfg')
+
+Write
+with open('example.ini', 'w') as configfile:
+...   config.write(configfile)
+"""
+
+# Globals
+SAMPLESHEET_HEADER_REGEX = r"^\[(\S+)\](,+)?"  # https://regex101.com/r/5nbe9I/1
+V2_SAMPLESHEET_HEADER_VALUES = {"Data": "BCLConvert_Data",
+                                "Settings": "BCLConvert_Settings"}
+
+
+def get_args():
+    """
+    Get arguments for the command
+    """
+    parser = argparse.ArgumentParser(description="Create samplesheets based on override cycles inputs."
+                                                 "Expects a v1 samplesheet as input and overridecycles and samples"
+                                                 "as inputs through comma-separated arrays")
+
+    # Arguments
+    parser.add_argument("--samplesheet-csv", required=True,
+                        help="Path to samplesheet csv")
+    parser.add_argument("--out-dir", required=False,
+                        help="Output directory for samplesheets, set to cwd if not specified")
+    parser.add_argument("--samples", required=False,
+                        help="Comma separated values to match override-cycles arg")
+    parser.add_argument("--override-cycles", required=False,
+                        help="Comma separated values to match samples arg")
```

Comment:
To confirm, if these arguments are "None", the same input samplesheet is returned?

Reply:
Yes, unless `--v2` is also specified, in which case the samplesheet headers are converted for `Settings` and `Data`

Reply:
And that same samplesheet is placed in the directory specified by `--out-dir` as `SampleSheet.csv`